tcmur_cmd_handler.c

```diff
@@ -216,23 +218,13 @@ static struct unmap_state *unmap_state_alloc(struct tcmu_device *dev,
 	return NULL;
 }
 
-static void unmap_state_free(struct unmap_state *state)
+static void unmap_state_put(struct tcmu_device *dev,
+			    struct tcmulib_cmd *origcmd, int ret)
 {
-	pthread_mutex_destroy(&state->lock);
-	free(state);
-}
-
-static void handle_unmap_cbk(struct tcmu_device *dev, struct tcmulib_cmd *ucmd,
-			     int ret)
-{
-	struct unmap_descriptor *desc = ucmd->cmdstate;
-	struct tcmulib_cmd *origcmd = desc->origcmd;
 	struct unmap_state *state = origcmd->cmdstate;
 	bool error;
 	int status;
 
-	free(desc);
-
 	pthread_mutex_lock(&state->lock);
 	error = state->error;
```

Comment:
With your fixes/cleanups in the patch, we can just drop state->error now too right?

Here we can do

```
if (state->status != TCMU_STS_OK && ret)
      state->status = ret;
```

Reply:
Yes, you can drop all that its not needed anymore after the correct status change. - Sorry I didn't clean it up before, I'm traveling atm. So i'm kinda rushed.